assets/js/modules/adsense/components/dashboard/AdSenseDashboardWidgetOverview.js

```diff
@@ -0,0 +1,188 @@
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../datastore/constants';
+import { Cell, Grid, Row } from '../../../../material-components';
+import PreviewBlock from '../../../../components/PreviewBlock';
+import DataBlock from '../../../../components/data-block';
+import getDataErrorComponent from '../../../../components/legacy-notifications/data-error';
+import getNoDataComponent from '../../../../components/legacy-notifications/nodata';
+import { readableLargeNumber, changeToPercent, numberFormat } from '../../../../util';
+import { isZeroReport } from '../../util';
+const { useSelect } = Data;
+
+export default function AdSenseDashboardWidgetOverview( props ) {
+	const {
+		startDate,
+		endDate,
+		compareStartDate,
+		compareEndDate,
+		metrics,
+		selectedStats,
+		handleStatSelection,
+		handleDataError,
+		handleDataSuccess,
+	} = props;
+
+	const currentRangeArgs = {
+		metrics: Object.keys( metrics ),
+		startDate,
+		endDate,
+	};
+
+	const previousRangeArgs = {
+		metrics: Object.keys( metrics ),
+		startDate: compareStartDate,
+		endDate: compareEndDate,
+	};
+
+	const currentRangeData = useSelect( ( select ) => select( STORE_NAME ).getReport( currentRangeArgs ) );
+	const previousRangeData = useSelect( ( select ) => select( STORE_NAME ).getReport( previousRangeArgs ) );
+
+	const currentDataLoaded = useSelect( ( select ) => select( STORE_NAME ).hasFinishedResolution( 'getReport', [ currentRangeArgs ] ) );
+	const previousDataLoaded = useSelect( ( select ) => select( STORE_NAME ).hasFinishedResolution( 'getReport', [ previousRangeArgs ] ) );
+
+	const currentError = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ currentRangeArgs ] ) );
+	const previousError = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ previousRangeArgs ] ) );
+
+	// TODO: remove the following logic when AdSenseDashboardWidget is refactored.
+	useEffect( () => {
+		if ( currentDataLoaded && previousDataLoaded ) {
+			if ( currentError || previousError ) {
+				handleDataError( currentError || previousError );
+			} else if ( isZeroReport( currentRangeData ) ) {
+				handleDataError();
+			} else {
+				handleDataSuccess();
+			}
+		}
+	}, [
+		currentDataLoaded && previousDataLoaded, // All reports are fetched.
+		!! currentError || !! previousError, // Whether there is an error or not.
+		JSON.stringify( currentRangeData ),
+	] );
+
+	if ( ! currentDataLoaded || ! previousDataLoaded ) {
+		return <PreviewBlock width="100%" height="250px" />;
+	}
+
+	if ( currentError || previousError ) {
+		const error = currentError || previousError;
+		return getDataErrorComponent( 'adsense', error.message, false, false, false, error );
+	}
+
+	if ( isZeroReport( currentRangeData ) ) {
+		return getNoDataComponent( _x( 'AdSense', 'Service name', 'google-site-kit' ), true, true, true );
```

Comment:
Can you replace this with using the new `ReportZero` component?

Reply:
Updated.